### ISSUE TITLE
ES readiness probe timing out

### ISSUE THREAD

**Issue by ewolinetz** [openshift/origin-aggregated-logging#456]
It seems that the time to wait on the curl and the probe configuration isn't enough time to give ES a chance to respond in the case that it is under load:
```
[root@host-8-172-177 ~]# oc exec logging-es-data-master-xpehwc99-4-n647k -- curl -X HEAD --cacert /etc/elasticsearch/secret/admin-ca --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key --max-time 4 -w '%{response_code}' https://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:04 --:--:--     0
curl: (28) Operation timed out after 4097 milliseconds with 0 out of 0 bytes received
000

[root@host-8-172-177 ~]# oc exec logging-es-data-master-xpehwc99-4-n647k -- curl -X HEAD --cacert /etc/elasticsearch/secret/admin-ca --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secretmin-key --max-time 5 -w '%{response_code}' https://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:05 --:--:--     0
curl: (28) Operation timed out after 5099 milliseconds with 0 out of -1 bytes received
000

[root@host-8-172-177 ~]# oc exec logging-es-data-master-xpehwc99-4-n647k -- curl -X HEAD --cacert /etc/elasticsearch/secret/admin-ca --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secretmin-key --max-time 30 -w '%{response_code}' https://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:22 --:--:--     0200

[root@host-8-172-177 ~]# oc exec logging-es-data-master-xpehwc99-4-n647k -- curl -X HEAD --cacert /etc/elasticsearch/secret/admin-ca --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key --max-time 30 -w '%{response_code}' https://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
200

[root@host-8-172-177 ~]# oc exec logging-es-data-master-xpehwc99-4-n647k -- curl -X HEAD --cacert /etc/elasticsearch/secret/admin-ca --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secretmin-key --max-time 30 -w '%{response_code}' https://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:18 --:--:--     0200

[root@host-8-172-177 ~]# oc exec logging-es-data-master-xpehwc99-4-n647k -- curl -X HEAD --cacert /etc/elasticsearch/secret/admin-ca --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key --max-time 30 -w '%{response_code}' https://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:06 --:--:--     0200

[root@host-8-172-177 ~]# oc exec logging-es-data-master-xpehwc99-4-n647k -- curl -X HEAD --cacert /etc/elasticsearch/secret/admin-ca --cert /etc/elasticsearch/secret/admin-cert --key /etc/elasticsearch/secret/admin-key --max-time 30 -w '%{response_code}' https://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:08 --:--:--     0200
```

The current max-time value is `4` however on most occasions it took at least 6 seconds to return (peaked at 22 in these tests)

We should be able to configure this value with an ENV var or it should be bumped up. This is preventing ES from being marked as ready for QE (https://bugzilla.redhat.com/show_bug.cgi?id=1456139)

**Comment by richm** [comment]
We should be able to configure this value from the command line and ENV var and it should be bumped up.

**Comment by jcantrill** [comment]
I think configuration is reasonable though we should also understand why it is taking so long.

**Comment by ewolinetz** [comment]
In this env it is a ES deployed with INSTANCE_RAM=512Mi

**Comment by ewolinetz** [comment]
Addressed by https://github.com/openshift/origin-aggregated-logging/pull/457

**Comment by jcantrill** [comment]
Fixed by #457
